### PR TITLE
Rename ZIPCode to PostalCode

### LINKED ISSUE #164: Rename ZIPCode logical type to PostalCode
`ZIPCode` has limited meaning outside of the United States, so this logical type should be renamed to something more generic such as `PostalCode` to have more global applicability.

### PR DESCRIPTION
**Pull request by thehomebrewnerd**
- Rename ZIPCode to PostalCode
- Closes #164 

Renames ZIPCode logical type to PostalCode. Also had to update one test because the renaming changed the order or output from the `ww.list_logical_types()` function.

### CODE REVIEW

File: woodwork/statistics_utils.py [hunk 1]
```diff
@@ -168,7 +168,7 @@ def _get_mutual_information_dict(dataframe, num_bins=10, nrows=None, include_ind
     """Calculates mutual information between all pairs of columns in the DataFrame that
     support mutual information. Logical Types that support mutual information are as
     follows:  Boolean, Categorical, CountryCode, Datetime, Double, Integer, Ordinal,
-    SubRegionCode, and ZIPCode
+    SubRegionCode, and PostalCode
```
Comment (tamargrey) [inline woodwork/statistics_utils.py:171]: I think we're trying to do alphabetical order here, so we should probably switch `SubRegionCode` and `PostalCode`

File: woodwork/table_accessor.py [hunk 2]
```diff
@@ -681,7 +681,7 @@ def mutual_information_dict(self, num_bins=10, nrows=None, include_index=False):
         Calculates mutual information between all pairs of columns in the DataFrame that
         support mutual information. Logical Types that support mutual information are as
         follows:  Boolean, Categorical, CountryCode, Datetime, Double, Integer, Ordinal,
-        SubRegionCode, and ZIPCode
+        SubRegionCode, and PostalCode
```
Comment (tamargrey) [inline woodwork/table_accessor.py:684]: here too

File: woodwork/table_accessor.py [hunk 3]
```diff
@@ -708,7 +708,7 @@ def mutual_information(self, num_bins=10, nrows=None, include_index=False):
         """Calculates mutual information between all pairs of columns in the DataFrame that
         support mutual information. Logical Types that support mutual information are as
         follows:  Boolean, Categorical, CountryCode, Datetime, Double, Integer, Ordinal,
-        SubRegionCode, and ZIPCode
+        SubRegionCode, and PostalCode
```
Comment (tamargrey) [inline woodwork/table_accessor.py:711]: here too (maybe it's a little silly that we have this docstring in three separate places)

**Review by tamargrey — APPROVED** [review]
small comment about alphabetical order, but lgmt other than that!

File: woodwork/statistics_utils.py (hunk 1 above)
Comment (thehomebrewnerd) [reply woodwork/statistics_utils.py:171]: Yep, I rearranged other places to keep the order, but missed this. I'll update these before merging.

**Pull request merged**